config/muxcable.py

```diff
@@ -73,43 +73,25 @@ def lookup_statedb_and_update_configdb(per_npu_statedb, config_db, port, state_c
     ipv6_value = get_value_for_key_in_config_tbl(config_db, port, "server_ipv6", "MUX_CABLE")
 
     state = get_value_for_key_in_dict(muxcable_statedb_dict, port, "state", "MUX_CABLE_TABLE")
-    if (state == "active" and configdb_state == "active") or (state == "standby" and configdb_state == "active") or (state == "unknown" and configdb_state == "active"):
-        if state_cfg_val == "active":
-            # status is already active, so right back error
-            port_status_dict[port] = 'OK'
-        if state_cfg_val == "auto":
-            # display ok and write to cfgdb auto
-            port_status_dict[port] = 'OK'
-            config_db.set_entry("MUX_CABLE", port, {"state": "auto",
-                                                    "server_ipv4": ipv4_value, "server_ipv6": ipv6_value})
-    elif state == "active" and configdb_state == "auto":
-        if state_cfg_val == "active":
-            # make the state active and write back OK
-            config_db.set_entry("MUX_CABLE", port, {"state": "active",
-                                                    "server_ipv4": ipv4_value, "server_ipv6": ipv6_value})
-            port_status_dict[port] = 'OK'
-        if state_cfg_val == "auto":
-            # dont write anything to db, write OK to user
-            port_status_dict[port] = 'OK'
 
-    elif (state == "standby" and configdb_state == "auto") or (state == "unknown" and configdb_state == "auto"):
-        if state_cfg_val == "active":
-            # make the state active
-            config_db.set_entry("MUX_CABLE", port, {"state": "active",
+    if str(state_cfg_val) == str(configdb_state):
+        port_status_dict[port] = 'OK'
+    else:
+        if str(state_cfg_val) == 'active' and str(state) != 'active':
+            config_db.set_entry("MUX_CABLE", port, {"state": state_cfg_val,
```

Comment:
the `config_db.set_entry` should be outside the if statement.

Reply:
fixed